core/src/labels/label.cpp

```diff
@@ -30,38 +32,80 @@ Label::Label(Label::Transform _transform, glm::vec2 _size, Type _type, Options _
 
 Label::~Label() {}
 
-bool Label::updateScreenTransform(const glm::mat4& _mvp, const glm::vec2& _screenSize, bool _drawAllLabels) {
-
-    glm::vec2 screenPosition;
+bool Label::updateScreenTransform(const glm::mat4& _mvp, const ViewState& _viewState, bool _drawAllLabels) {
```

Comment:
@hjanetzek One last issue I just found; when used as markers, the `mvp` matrix misses a scale component since the point marker has no extent, so the projection to screen goes wrong.